src/main/java/sixgaezzang/sidepeek/skill/serivce/SkillService.java

```diff
@@ -0,0 +1,42 @@
+package sixgaezzang.sidepeek.skill.serivce;
+
+import static sixgaezzang.sidepeek.common.ValidationUtils.validateMaxLength;
+
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import sixgaezzang.sidepeek.skill.domain.Skill;
+import sixgaezzang.sidepeek.skill.dto.SkillResponse;
+import sixgaezzang.sidepeek.skill.dto.SkillSearchResponse;
+import sixgaezzang.sidepeek.skill.repository.SkillRepository;
+
+@Service
+@RequiredArgsConstructor
+public class SkillService {
+
+    public static final int KEYWORD_MAX_LENGTH = 50;
+
+    private final SkillRepository skillRepository;
+
+    public SkillSearchResponse searchByName(String keyword) {
+        List<Skill> skills;
+        if (Objects.isNull(keyword) || keyword.isBlank()) {
+            skills = skillRepository.findAll();
+        } else {
+            validateMaxLength(keyword, KEYWORD_MAX_LENGTH,
+                "최대 " + KEYWORD_MAX_LENGTH + "자의 키워드로 검색할 수 있습니다.");
+            skills = skillRepository.findAllByNameContaining(keyword);
+        }
+
+        List<SkillResponse> searchResults = skills.stream()
+            .map(skill -> SkillResponse.builder()
+                .id(skill.getId())
+                .name(skill.getName())
+                .iconImageUrl(skill.getIconImageUrl())
+                .build()
+            ).toList();
```

Comment:
만약 `SkillResponse`에 `Skill`을 받는 정적 팩터리 메서드를 추가한다면, 
코드가 간결해 질수도 있을 것 같아용!

```suggestion
        List<SkillResponse> searchResults = skills.stream()
            .map(SkillResponse::from)
            .toList();
```

Reply:
역시 의진님.. 👍 정적 팩터리 메서드 적극적으로 활용하는 거 좋은 것 같습니다!

Reply:
너무 좋습니다! 요것도 위치가 애매해서 이야기 해보렷했ㄴ는데 우선 dto쪽에 메서드 넣는 것으로 하겠습니다!